test/phx_tools_web/plugs/curl_detector_test.exs

```diff
@@ -0,0 +1,27 @@
+defmodule PhxToolsWeb.CurlDetectorTest do
+  use PhxToolsWeb.ConnCase, async: true
+
+  @script_content "#!/bin/sh"
+
+  describe "call/2" do
+    test "returns script content for requests with curl User-Agent on root path", %{conn: conn} do
+      conn =
+        conn
+        |> put_req_header("user-agent", "curl/7.68.0")
+        |> get("/")
+
+      assert conn.status == 200
+      assert String.contains?(conn.resp_body, @script_content)
+    end
+
+    test "does not return script content for requests with non-curl User-Agent", %{conn: conn} do
+      conn =
+        conn
+        |> put_req_header("user-agent", "Mozilla/5.0")
+        |> get("/macOS.sh")
```

Comment:
Use the same path as above. Otherwise, the test could pass even if the plug's implementation is incorrect.

Let me explain.
Look at the order of the plugs in the `PhxToolsWeb.Endpoint`.
https://github.com/optimumBA/phx.tools/blob/2ce95cbdf62c8ab0bae466664439a0f6278804b6/lib/phx_tools_web/endpoint.ex#L20-L24
https://github.com/optimumBA/phx.tools/blob/2ce95cbdf62c8ab0bae466664439a0f6278804b6/lib/phx_tools_web/endpoint.ex#L48
Static files take priority over the plugs and routes we defined in the `PhxToolsWeb.Router`. That means even if you've had
```elixir
def call(conn, _opts) do
  conn
  |> put_resp_content_type("text/plain")
  |> send_resp(200, script_content())
  |> halt()
end
```
in your plug, your tests would still pass because the plug's code isn't executed for the `/macOS.sh` path. Try it locally.

Also, you can see more quickly by running `mix test --cover test/phx_tools_web/plugs/curl_detector_test.exs`. You'll see that the plug is not 100% covered.

You can leave the tests for `/Linux.sh` and `/macOS.sh` here, but make sure there's also a test for `/` when the user-agent is not `curl`. For the `/Linux.sh` and `macOS.sh` continue using `Mozilla` so we're sure we still serve files even when the user agent is not `curl`.